File: 1-two-sum/two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        d={}
        res=[]
        for i in range(len(nums)):
            d[nums[i]]=i
        for i in range(len(nums)):
            x=target-nums[i]
            if x==nums[i] and i!=d[nums[i]]:
                res.append(i)
                res.append(d[nums[i]])
                break
            if x in nums:
                v=nums.index(x)
                print(v)
                if(v>i):
                    res.append(i)
                    res.append(nums.index(x))
                    break
        return res

        
                

            

File: 1-two-sum/two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer,Integer> h =new HashMap<>();
        List<Integer> list=new ArrayList<>();
        for(int i:nums){
            list.add(i);
        }
        for(int i=0;i<nums.length;i++){
            h.put(nums[i],i);
        }
        int[] res=new int[2];
        for(int i=0;i<nums.length;i++){
            int x=target-nums[i];
            if(list.contains(x) && list.indexOf(x)!=i){
                res[0]=i;
                res[1]=list.indexOf(x);
                return res;
            }
        }
        return new int[0];
        
        
    }
}